metagraph/src/common/elias_fano.cpp

```diff
@@ -174,11 +174,13 @@ template <typename T>
 EliasFanoEncoder<T>::EliasFanoEncoder(size_t size,
                                       T min_value,
                                       T max_value,
-                                      const std::string &out_filename)
+                                      const std::string &out_filename,
+                                      bool append)
     : declared_size_(size), offset_(min_value) {
-    sink_internal_ = std::ofstream(out_filename, std::ios::binary);
+    auto mode = append ? std::ios::binary | std::ios::app : std::ios::binary;
+    sink_internal_ = std::ofstream(out_filename, mode);
     sink_ = &sink_internal_;
-    sink_internal_upper_ = std::ofstream(out_filename + ".up", std::ios::binary);
+    sink_internal_upper_ = std::ofstream(out_filename + ".up", mode);
```

Comment:
Just to make it clear it's always binary

```suggestion
    sink_internal_upper_ = std::ofstream(out_filename + ".up", mode | std::ios::binary);
```